src/hasSubObject.test.ts

```diff
@@ -96,15 +96,35 @@ describe("typing", () => {
       const obj = {} as { a?: string; b?: number };
 
       if (hasSubObject(obj, {})) {
+        expectTypeOf(obj).toEqualTypeOf<
+          HasSubObjectBrand & { a?: string; b?: number }
```

Comment:
Use `toMatchTypeOf` and don't include the brand.
```suggestion
        expectTypeOf(obj).toMatchTypeOf<
          { a?: string; b?: number }
```